book/src/types/enums.md

```diff
@@ -1,73 +1,151 @@
+# extern crate juniper;
+# use juniper::GraphQLEnum;
+#
+#[derive(GraphQLEnum)]
+#[graphql(name = "WrongEpisode")] // now exposed as `WrongEpisode` in GraphQL schema
+enum Episode {
+    #[graphql(name = "LAST_HOPE")]
+    NewHope, // exposed as `LAST_HOPE` in GraphQL schema
     Empire,
     Jedi,
 }
+#
+# fn main() {}
+```
 
+Or provide a different renaming policy for all the [enum][3] variants:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLEnum;
+#
+#[derive(GraphQLEnum)]
+#[graphql(rename_all = "none")] // disables any renaming
+enum Episode {
+    NewHope, // exposed as `NewHope` in GraphQL schema
+    Empire,  // exposed as `Empire` in GraphQL schema
+    Jedi,    // exposed as `Jedi` in GraphQL schema
+}
+#
 # fn main() {}
 ```
+> **TIP**: Supported policies are: `SCREAMING_SNAKE_CASE`, `camelCase` and `none` (disables any renaming).
 
-## Documentation and deprecation
 
-Just like when defining objects, the type itself can be renamed and documented,
-while individual enum variants can be renamed, documented, and deprecated:
+### Documentation and deprecation
 
+Just like when [defining GraphQL objects](objects/index.md#documentation), the [GraphQL enum][0] type and its values could be [documented][4] and [deprecated][5] via `#[graphql(description = "...")]` and `#[graphql(deprecated = "...")]`/[`#[deprecated]`][13] attributes:
 ```rust
 # extern crate juniper;
-#[derive(juniper::GraphQLEnum)]
-#[graphql(name="Episode", description="An episode of Star Wars")]
+# use juniper::GraphQLEnum;
+#
+/// This doc comment is visible only in Rust API docs.
+#[derive(GraphQLEnum)]
+#[graphql(description = "An episode of Star Wars")]
 enum StarWarsEpisode {
-    #[graphql(deprecated="We don't really talk about this one")]
-    ThePhantomMenace,
-
-    #[graphql(name="NEW_HOPE")]
+    /// This doc comment is visible only in Rust API docs.
+    #[graphql(description = "This description is visible only in GraphQL schema.")]
     NewHope,
 
-    #[graphql(description="Arguably the best one in the trilogy")]
+    /// This doc comment is visible only in Rust API docs.
+    #[graphql(desc = "Arguably the best one in the trilogy.")]
+    //        ^^^^ shortcut for a `description` argument
     Empire,
+
+    /// This doc comment is visible in both Rust API docs and GraphQL schema 
+    /// descriptions.
     Jedi,
+    
+    #[deprecated(note = "Only visible in Rust.")]
+    #[graphql(deprecated = "We don't really talk about this one.")]
+    //        ^^^^^^^^^^ takes precedence over Rust's `#[deprecated]` attribute
+    ThePhantomMenace, // has no description in GraphQL schema
 }
+#
+# fn main() {}
+```
+> **NOTE**: Only [GraphQL object][6]/[interface][7] fields and [GraphQL enum][0] values could be [deprecated][5].
+
 
+### Ignoring
+
+By default, all [enum][3] variants are included into the generated [GraphQL enum][0] type as values. To prevent including a specific variant, it should be annotated with the `#[graphql(ignore)]` attribute:
```

Comment:
```suggestion
By default, all [enum][3] variants are included in the generated [GraphQL enum][0] type as values. To prevent including a specific variant, annotate it with the `#[graphql(ignore)]` attribute:
```